[VideoPlayerVideo] Dont require extradata for avc3, avc4

CastagnaIT opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
On the note of "Chapter 3. Initialization Segments" of ISOBMFF
https://www.w3.org/TR/mse-byte-stream-format-isobmff/#iso-init-segments
say:
`The user agent must support parameter sets (e.g., PPS/SPS) stored in the sample entry (as defined for avc1/avc2), and should support parameter sets stored inband in the samples themselves (as defined for avc3/avc4).`

so the initialization packet for avc1/avc2 contains PPS/SPS that we can retrieve with ISAdaptive demuxer and set it as extradata
instead in the avc3/avc4 variants there is no PPS/SPS in the initialization packet so the extradata member will be empty

kodi with `hint.codec == AV_CODEC_ID_H264` assume that all h264 variants require extradata
so this add exception for avc3/avc4

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
fix video playback with DRM on Android OS

fix https://github.com/xbmc/inputstream.adaptive/issues/1511

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
a first rougly test confirmed that works

and user confirmed works https://github.com/xbmc/inputstream.adaptive/issues/1511#issuecomment-2028050892

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->
can play h264 avc3/avc4 DRM videos on android

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

nirvana-7777 commented:
I wonder if would make more sense to make avc1/2 the exception and avc3/4 the new default behaviour in order to be more future-proof.

CastagnaIT commented:
> I wonder if would make more sense to make avc1/2 the exception and avc3/4 the new default behaviour in order to be more future-proof.

the problem is that you are referring to a "hint" data, and the hint data is somewhat optional
you cannot be sure that all demuxers provide codec_tag hint data, so IMO better check for avc3/avc4

nirvana-7777 commented:
That's what I meant. Anyway it's speculation on what the future is more likely to be...

CastagnaIT committed:
[VideoPlayerVideo] Dont require extradata for avc3, avc4

for avc3, avc4 variants extradata is stored on the sample packets and not in the
initialization packet

CastagnaIT commented on xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp:
```diff
@@ -112,7 +112,7 @@ bool CVideoPlayerVideo::OpenStream(CDVDStreamInfo hint)
         hint.codec == AV_CODEC_ID_MPEG1VIDEO ||
         hint.codec == AV_CODEC_ID_MPEG2VIDEO ||
-        hint.codec == AV_CODEC_ID_H264 ||
+        (hint.codec == AV_CODEC_ID_H264 && (hint.codec_tag != MKTAG('a','v','c','3') && hint.codec_tag != MKTAG('a','v','c','4'))) ||
```
condition has been changed

CastagnaIT commented:
user have tested and confirmed that the change works

CastagnaIT commented:
jenkins build this please
